cram_external_interfaces/kdl_ik_service/src/kdl_ik_service/ik.py

```diff
@@ -103,10 +105,15 @@ def get_joint_limits_from_urdf(joint_name, urdf_obj):
         [joint_found] = [joint for joint in urdf_joints if joint.name == joint_name]
     except ValueError as e:
         raise ValueError("Error while trying to find joint named: %s in the urdf. Reason: %s" % (joint_name, e))
-    if joint_found.joint_type != 'fixed':
-        return joint_found.limit.upper, joint_found.limit.lower
-    else:
+    # Continuous joints (0 as joint limits will mess up the IK solution)
+    if joint_found.joint_type == 'continuous' and joint_found.limit.upper == joint_found.limit.lower == 0:
+        return 2 * math.pi, -2 * math.pi
```

Comment:
Actually, I dont understand why you have to specifically check if the limits are 0.
Shouldn't the limits of any continuous joint be -2pi; 2pi?
I tried this (just commented out the if of the limits == 0) and got the same weird behaviour of not converging, that you said was caused by -4pi; 4pi limits.
Then I changed the limits from -2pi; 2pi into -pi; pi, and the IK ended up having a lot of difficulty in finding solutions.

Why did I even play with this?
The IK for the right arm doesn't really work.
@Tigul  tried the mirror poses of the following:
https://hackmd.informatik.uni-bremen.de/lpAGpTLhQXWgIFIH4hDHow#

@amar-fayaz could you please take a look at this again.